### ISSUE TITLE
Feature Request: Overwrite previous logged values if a run is resumed

### ISSUE THREAD

**Issue by neerajwagh** [neptune-ai/neptune-client#1330]
### Is your feature request related to a problem? Please describe.

Consider an example training run:
- The run lasts 500 epochs.
- A model checkpoint is saved every 10 epochs (since saving models after every epoch wastes disk space) but the train/val loss is logged/appended to the run after every epoch. 

The problem:
When resuming model training, the latest checkpoint is at epoch(=step_idx) 490, but the train/val loss logged to the Neptune run is logged at step_idx=499.

This throws the error:
`Error occurred during asynchronous operation processing: X-coordinates (step) must be strictly increasing for series attribute: train/loss. Invalid point: 490.0 (or 491.0 depending on how you resume counting epochs)
`

Model training is technically resuming at epoch 490/491 but Neptune refuses to over-write the previously logged values and logs nothing instead. I suspect Neptune will keep throwing this error and log nothing until the resumed run crosses the 499 step_idx mark (edit: yep that's exactly what happens:)

```
Error occurred during asynchronous operation processing: X-coordinates (step) must be strictly increasing for series attribute: val/loss. Invalid point: 497.0
[TRAIN: 498] loss after epoch 498: 6.235099507672956
[VAL: 498] loss after epoch 498: 9.273474887925751
Error occurred during asynchronous operation processing: X-coordinates (step) must be strictly increasing for series attribute: train/loss. Invalid point: 498.0
Error occurred during asynchronous operation processing: X-coordinates (step) must be strictly increasing for series attribute: val/loss. Invalid point: 498.0
[TRAIN: 499] loss after epoch 499: 6.2347105841930786
Error occurred during asynchronous operation processing: X-coordinates (step) must be strictly increasing for series attribute: train/loss. Invalid point: 499.0
[VAL: 499] loss after epoch 499: 10.536410234412369
Error occurred during asynchronous operation processing: X-coordinates (step) must be strictly increasing for series attribute: val/loss. Invalid point: 499.0
[TRAIN: 500] loss after epoch 500: 6.2348618278717725
[VAL: 500] loss after epoch 500: 11.13005256652832
[TRAIN: 501] loss after epoch 501: 6.23495394964527
[VAL: 501] loss after epoch 501: 10.467232976640974
[TRAIN: 502] loss after epoch 502: 6.2348632253442435
[VAL: 502] loss after epoch 502: 9.382444595804019
[TRAIN: 503] loss after epoch 503: 6.234755640149192
[VAL: 503] loss after epoch 503: 15.499999941611776
[TRAIN: 504] loss after epoch 504: 6.234689587749048
[VAL: 504] loss after epoch 504: 8.723198384654765
[TRAIN: 505] loss after epoch 505: 6.234622372752921
[VAL: 505] loss after epoch 505: 9.153765834107691
[TRAIN: 506] loss after epoch 506: 6.234649834797432
[VAL: 506] loss after epoch 506: 8.68311222232118
```

### Describe the solution you'd like

Overwrite the old logged values if the run is resumed at a previous step idx.

### Describe alternatives you've considered
Forcefully log the model after the last epoch of training even if it breaks the preset model save frequency. This will ensure the step_idx of Neptune and latest saved model are in sync when the run is resumed.

**Comment by Blaizzy** [comment]
Hi @neerajwagh 

You can overwrite metrics by downloading the metric from neptune, which will come in a Data Frame format, and prune it to the desired epoch/step.

```python
import neptune
run = neptune.init_run(with_id='ABC-123')

metrics = run['metrics'].fetch()

epoch = 10
pruned_metrics = metrics[metrics['step'] < epoch] # metrics before epoch 10

del run['metrics']

run['metrics'].extend(pruned_metrics) # recreate the metrics chart with pruned metrics 

# Continue logging
```

Docs: https://docs.neptune.ai/api/field_types/#floatseries

**Comment by SiddhantSadangi** [comment]
Hello @neerajwagh ,
Just checking if the above solution worked for you :)

**Comment by Blaizzy** [comment]
I'm closing the issue for now. 
Please feel free to comment; we will pick up from where we left off.
